nsds_lab_to_nwb/metadata/metadata_manager.py

```diff
@@ -1,13 +1,126 @@
+import logging
 import os
 import csv
 import pandas as pd
 from ..utils import (get_metadata_lib_path, get_stim_lib_path,
                      split_block_folder)
 
-from nsds_lab_to_nwb.common.io import read_yaml, csv_to_dict
+from nsds_lab_to_nwb.common.io import read_yaml, write_yaml, csv_to_dict
 
 
-_DEFAULT_EXPERIMENT_TYPE = 'auditory'  # for legacy sessions
+_DEFAULT_EXPERIMENT_TYPE = 'auditory'
+_WRITE_YAML_FILE = True     # export intermediate yaml files for metadata inspection
+                            # turn off when testing is over
+
+logger = logging.getLogger(__name__)
+
+
+class MetadataReader:
+    ''' Reads metadata input for new experiments.
+    '''
+    def __init__(self,
+                block_metadata_path: str,
+                library_path: str,
+                block_folder: str,
+                ):
+        self.block_metadata_path = block_metadata_path
+        self.library_path = library_path
+        self.block_folder = block_folder
+
+    def read(self, write_yaml_file=_WRITE_YAML_FILE):
+        self.metadata_input = self.load_metadata_source()
+        if write_yaml_file:
+            write_yaml(f'_test/{self.block_folder}_metadata_input.yaml',
+                       self.metadata_input)
+
+        self.extra_cleanup()
+        if write_yaml_file:
+            write_yaml(f'_test/{self.block_folder}_metadata_input_clean.yaml',
```

Comment:
Should this depend on `save_path`?

Reply:
now using `metadata_save_path` (which defaults to None, and if None the file is not saved)